### ISSUE TITLE
Document params in al_document.py more

### ISSUE THREAD

**Issue by BryceStevenWilley** [SuffolkLITLab/docassemble-AssemblyLine#626]
For example, in the [`as_zip` function, `title` doesn't seem to do anything](https://github.com/SuffolkLITLab/docassemble-AssemblyLine/blob/45e4ae4ecba561717d26cc89f77630825e3415e4/docassemble/AssemblyLine/al_document.py#L1067). When looking around a bit more, it seems that it's coupled to `download_list_html`, where it's used in the HTML table row, but the param isn't used there, and it isn't clear that's what the param is used for.

Probably are other params like that in the file, we should fix that and a few other ones as well.

**Comment by nonprofittechy** [comment]
This is a pretty vague issue. Can keep it open for now but suggest we put a timeline on it or close it.

**Comment by nonprofittechy** [comment]
Maybe adequately addressed by #734?

**Comment by BryceStevenWilley** [comment]
I don't really think #734 addresses this; it adds docstrings for things, but those aren't great. For example, for `as_zip(title="my title"...)`; the docstring now says  "Title of the zip file", but what does that mean? Zip files don't have titles as a part of the file or anything. It's just not documented that it's an attribute that's used by a different function, `download_list_html`.

But that function doesn't use the `title` parameter at all. So now, we have a parameter that's documented, and looks like it does something, but if you wanted to use it, you'd be thoroughly lost and not know how to use it, as most people aren't calling `as_zip` directly, and there's no passed-through param in `download_list_html`. The only solution is to use the class attribute, but it's all a little confusing to deduce, even looking at the code.

The reason I made the issue is that we want people to look at our reference code API docs more, and to do that, we need specific and actionable documentation for important params. After being fairly confused by `title`, I realized that many of our parameters would be like this for people. I guess "better documentation" isn't that actionable, but I did some of it in https://github.com/SuffolkLITLab/docassemble-AssemblyLine/pull/743.
